### ISSUE TITLE
validate the resulting file

### ISSUE THREAD

**Issue by jkowalleck** [CycloneDX/cyclonedx-php-composer#96]
:athletic_shoe:  this feature is currently in progress. 

idea: 
validate resulting XML/JSON

how to do it:

- [x]  ship the supported schemata files in the `res` folder in a way they dont need any internet connection to download additional files - done via #90 
- write a validator class
  - [ ] refactor existing schema validators in the integration-tests therefore
  - [ ] apply it in the integration tests, too
  - [ ] for XML 
  - [ ] for JSON based on the strict schema 
- [ ] add a CLI switch: `--no-validate` to diable validation, but have validation enabled per default
- add validator to the command
  - [ ] if validation enabled: validate the result based on shipped offline files (test with network disabled)
  - [ ] if validation failes: prompt an error message, show link to "create issue" and ask for the `composer.lock` file
- [ ]  validation is done on the serialized string, before any file is written.
- [ ] validation is not part of the serialization itself, but an extra step.

**Comment by jkowalleck** [comment]
JSON validator is blocked by https://github.com/CycloneDX/specification/issues/83
